openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/ORSGraphStorageFactory.java

```diff
@@ -86,9 +86,6 @@ else if (geTurnCosts != null && !graphExtensions.isEmpty()) {
 				((ORSGraphHopper) gh).initCoreLMAlgoFactoryDecorator();
 		}
 
-		if (gh.getLMFactoryDecorator().isEnabled())
```

Comment:
Why was this removed?

Reply:
Fair point, this is quite obscure. This change was complemented by https://github.com/GIScience/graphhopper/commit/b44430e3ded517b4f161bd5ff16e7c495c04a1a1 and the rearrangement was necessary in order to get things to work. Unfortunately, I don't remember the exact details anymore, but if you follow the program flow then you will realize that before this change we had to call `initLMAlgoFactoryDecorator();` only because the original call in GH never happened because of this very modification in the fork:

```//ORS-GH MOD START
        if (graphStorageFactory != null) {
            ghStorage = graphStorageFactory.createStorage(dir, this);
        }

        if (ghStorage == null) {
        //ORS-GH MOD END```

Reply:
Okay, seems reasonable to me. The order of initialization is also kept intact. As long as the code in ORSGraphStorageFactory that follow the initialization of core and corelm does not depend on LM, I do not see a reason why we should not do it on the "outer layer". But why is the CH part still in this factory and not in the graphhopper code?

Reply:
The CH part has never been a subject to time-dependent routing and as such has not been modified in these PRs. 

The change to LM creation was necessary because of the fact that currently `TimeDependentFastestWeighting` requires `GraphHopperStorage` to be initialized beforehand in order to create the `SpeedCalculator`.